snapcraft/internal/build_providers/_qemu/_keys.py

```diff
@@ -0,0 +1,61 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from typing import TypeVar, Type
+
+from paramiko import RSAKey
+
+from snapcraft.internal.errors import SnapcraftError
+
+
+SSHKeyT = TypeVar('SSHKeyT', bound='SSHKey')
+
+
+class SSHKeyPathError(SnapcraftError):
+
+    fmt = (
+        '{private_key_file_path!r} does not exist. '
+        'A private key is required.\n'
+        'Please file a report on https://launchpad.net/snapcraft/+filebug'
+    )
+
+    def __init__(self, *, private_key_file_path: str) -> None:
+        super().__init__(private_key_file_path=private_key_file_path)
+
+
+class SSHKey:
```

Comment:
Class and function docs would be nice.

Reply:
hmm, I wonder where they went, I did actually write them :confused:

Reply:
reflog did not rescue me :-/